### PR TITLE
Add fallback sorting to SamplingReportDataView

### PR DESCRIPTION
**Pull request by beckerhe**
SamplingReportDataView used to perform stable_sort. stable_sort's outcome
depends on the previous state of the container. Since recent changes
trigger a reconstruction of SamplingReportDataView stable_sort
preserving behaviour doesn't work anymore since the container's previous
state is not preserved. That lead's to reorderings observed by the user.

This can be avoided by establishing a total ordering. In this commit I
use the function address as a unique and totally ordered field.

Test: Manual UI test
Bug: http://b/167541131

### CODE REVIEW

**Review by florian-kuebler — APPROVED** [review]
LGTM thanks. Just one minor comment regarding the commit message:
I guess we are using "Bug: " instead of "Fixes: ".

File: OrbitGl/SamplingReportDataView.cpp [hunk 1]
```diff
@@ -113,9 +113,29 @@ void SamplingReportDataView::DoSort() {
...
   }
 
-  if (sorter) {
-    std::stable_sort(indices_.begin(), indices_.end(), sorter);
+  if (!sorter) {
+    return;
   }
+
+  const auto fallback_sorter = [&](const auto& ind_left, const auto& ind_right) {
+    // address is the absolute function address. Hence it is unique and qualifies
```
Comment (florian-kuebler) [inline OrbitGl/SamplingReportDataView.cpp:121]: Also start with capital letters in Comments.

Comment (beckerhe) [reply OrbitGl/SamplingReportDataView.cpp:121]: Well, I used "address" because I meant the field name. I will add backticks to make that clear.

**Review by IrinaShkviro — APPROVED** [review]
Thanks for the fix!

File: OrbitGl/SamplingReportDataView.cpp (hunk 1 above)
Comment (florian-kuebler) [reply OrbitGl/SamplingReportDataView.cpp:121]: Thanks.

**Comment by beckerhe** [comment]
> LGTM thanks. Just one minor comment regarding the commit message:
> I guess we are using "Bug: " instead of "Fixes: ".

Ah thanks. I mixed this up. Should be fixed now.

**Comment by florian-kuebler** [comment]
Yes, looks good. Thanks!

**Pull request merged**